Add StitchCut macro for sheet metal stitch cuts

JohnHyslop opened the pull request:
This PR adds the StitchCut macro by John Hyslop.
It creates evenly spaced stitch cuts in sketches for hand-folded sheet metal parts.
Source repository: https://github.com/JohnHyslop/MyFreeCADMacros

JohnHyslop committed:
Create README.md

JohnHyslop committed:
Add files via upload

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

JohnHyslop committed:
[StitchCut] Clarify license to allow commercial use

luzpaz commented:
Is there any benefit to adding this to the [SheetMetal Workbench](https://github.com/shaise/FreeCAD_SheetMetal) ?

JohnHyslop commented:
> Is there any benefit to adding this to the [SheetMetal Workbench](https://github.com/shaise/FreeCAD_SheetMetal) ?
Yes, I feel it would be useful in the SheetMetal WB but it could also be useful for square and rectangular knockouts in panels.

JohnHyslop committed:
[StitchCut] Update macro metadata and version

- Added version 1.0
- License updated to include commercial use
- Metadata updated for Addon Manager compatibility

galou reviewed (changes requested):
Thanks for you contributions. Please react to my comments.

JohnHyslop commented on StitchCut/StitchCut.py:
```diff
@@ -0,0 +1,158 @@
+    and number of cuts. The macro will calculate the cut lengths and insert them directly
+    into the sketch.
+
+    Created with a little extra help from ChatGPT.
```
Ok no problem, I'll removed the reference.

JohnHyslop commented on StitchCut/StitchCut.py:
```diff
@@ -0,0 +1,158 @@
+
+import FreeCAD, FreeCADGui
+from PySide import QtGui
+import math, Part
```
Done.

JohnHyslop commented on StitchCut/StitchCut.py:
```diff
@@ -0,0 +1,158 @@
+    edge_offset, gap, num_cuts, cut_length = values
+
+    # Remove original line
+    sk.delGeometry(idx)
```
Transaction management has been added, allowing users to undo the macro actions in FreeCAD.

JohnHyslop reviewed (commented):
This addresses most of your comments.

galou commented:
> Once I figure out how to place the macro on GitHub as SheetMetal/StitchCut.FCMacro, I'll commit all the changes.

You just make the modification in your repo, commit and push to the same branch. The PR is automatically updated.

JohnHyslop commented on StitchCut/StitchCut.py:
```diff
@@ -0,0 +1,158 @@
+        ptB = FreeCAD.Vector(p1.x + direction.x*end_dist,
+                             p1.y + direction.y*end_dist,
+                             p1.z + direction.z*end_dist)
+        sk.addGeometry(Part.LineSegment(ptA, ptB), False)
```
Thanks for the suggestion! I’ll take a look at FreeCAD's preference system when I get a chance. I’m not very familiar with it, so I might need a bit of guidance on how to implement it properly, but I’ll give it a try.

Just out of curiosity, is there a specific reason the JSON method wouldn't be ideal? I'm happy to dive deeper into it if needed.

Cheers,
John

JohnHyslop commented:
Hi Galou,

I’ve made the requested changes and updated the SheetMetal/StitchCut macro, including the renaming, the icon, and the updated features. The wiki page has also been created with the relevant details, though I am currently experiencing an issue where the page displays as blank when opened. I’m looking into it, but everything is saved on my end.

Thanks for your help and suggestions!

Cheers,
John Hyslop

galou commented:
> If everything looks good to you

Hi @JohnHyslop, I can have missed something but the last changes I see are from Dec. 2025. To update the PR, just push commits to the same branch on your fork.

JohnHyslop committed:
Merge branch 'FreeCAD:master' into master

galou commented:
@JohnHyslop The only issue is that I cannot find your changes. Did you `git push`?

JohnHyslop committed:
Move StitchCut macro to SheetMetal folder, rename to SheetMetalStitchCut.FCMacro, update README and icon, bump version to 1.2

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

galou reviewed (approved):
Thanks for your contribution!

I still see a few issues that I'll correct myself.

galou commented:
Fixed in 61ecbc82e867.
